generators/app/templates/.gitignore

```diff
@@ -1,3 +1,35 @@
+################################################
+# macOS
```

Comment:
I didn't include the public stuff because we need those for a craft project but we haven't built out the craft side of this generator yet. 

The dist file locations felt a little more volitile than macOS files. 

I also grabbed this from that big ass repo full of common `.gitignores` which is why it's a little overbearing. I'm trusting in Open Source.